library/runtime/src/commonMain/kotlin/io/matthewnelson/kmp/tor/runtime/internal/TorProcess.kt

```diff
@@ -0,0 +1,564 @@
+import io.matthewnelson.kmp.tor.runtime.RuntimeEvent.Notifier.Companion.lce
+import io.matthewnelson.kmp.tor.runtime.RuntimeEvent.Notifier.Companion.p
+import io.matthewnelson.kmp.tor.runtime.RuntimeEvent.Notifier.Companion.w
+import io.matthewnelson.kmp.tor.runtime.TorRuntime
+import io.matthewnelson.kmp.tor.runtime.core.TorConfig
+import io.matthewnelson.kmp.tor.runtime.core.address.ProxyAddress
+import io.matthewnelson.kmp.tor.runtime.core.address.ProxyAddress.Companion.toProxyAddressOrNull
+import io.matthewnelson.kmp.tor.runtime.core.apply
+import io.matthewnelson.kmp.tor.runtime.core.ctrl.TorCmd
+import io.matthewnelson.kmp.tor.runtime.ctrl.TorCtrl
+import io.matthewnelson.kmp.tor.runtime.internal.TorProcess.StartArgs.Companion.createStartArgs
+import kotlinx.coroutines.*
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.concurrent.Volatile
+import kotlin.coroutines.cancellation.CancellationException
+import kotlin.jvm.JvmSynthetic
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.TimeSource
+
+internal class TorProcess private constructor(
+    private val generator: TorConfigGenerator,
+    private val NOTIFIER: RuntimeEvent.Notifier,
+    private val scope: CoroutineScope,
+    private val state: FIDState,
+): FileID by generator {
+
+    @Throws(Throwable::class)
+    private suspend fun <T: Any?> start(
+        connect: suspend CtrlArguments.() -> T
+    ): T = state.lock.withLock {
+        state.cancelAndJoinOtherProcess()
+
+        val (config, paths) = try {
+            generator.generate(NOTIFIER)
+        } catch (t: Throwable) {
+            NOTIFIER.lce(Lifecycle.Event.OnDestroy(this))
+            throw t
+        }
+
+        val startArgs = try {
+            config.createStartArgs(generator.environment)
+        } catch (t: Throwable) {
+            NOTIFIER.lce(Lifecycle.Event.OnDestroy(this))
+            throw t
+        }
+
+        // Want to use the path here and not the file b/c
+        // kmp-tor user could be using runtime to control
+        // a system installed tor and pass "tor".toFile()
+        // which is not absolute and, as such, Process will
+        // search for the tor program via PATH environment
+        // variable.
+        Process.Builder(command = paths.tor.path)
+            .args("--verify-config")
+            .args(startArgs.cmdLine)
+            .output{
+                timeoutMillis = 500
+            }.let { output ->
+                val validMsg = "Configuration was valid"
+                if (output.stdout.contains(validMsg, ignoreCase = true)) {
+                    NOTIFIER.i(this, validMsg)
+                } else {
+                    NOTIFIER.lce(Lifecycle.Event.OnDestroy(this))
+                    throw output.toInvalidConfigurationException()
+                }
+            }
```

Comment:
`output` can throw `IOException`, should be `try/catch`, dispatch LCE, then re-throw.